### DISCUSSION TITLE
Grandfather rollup via OpportunityContactRole not working

### DISCUSSION THREAD

**Discussion by marapetro** [jamessimone/apex-rollup#561] [General]
Hi James,

Thank you so much for building Apex Rollup! I am in the process of migrating my rollups and everything else has worked fine, but I am getting stuck with setting up a rollup from Opportunity (most recent (max) Stage_2_date__c) to Contact (Partner_Stage_2_Date__c) via OpportunityContactRole. The issue seems to be with defining the Grandparent Relationship Field Path. 
I've tried using 'OpportunityContactRole.Contact.Partner_Stage_2_Date__c' and 'OpportunityContactRoles.Contact.Partner_Stage_2_Date__c', but I am getting the apex error saying 'Didn't understand relationship 'OpportunityContactRole' in field path. If you are attempting to use a custom relationship, be sure to append the '__r' after the custom relationship name.'

Can you please help me understand what I am doing wrong?

Thanks,

Marija
![Screenshot 2024-02-13 at 3 37 07 PM](https://github.com/jamessimone/apex-rollup/assets/146372655/39304cb2-0f77-4ed3-b622-ddef7592dada)
<img width="870" alt="Screenshot 2024-02-13 at 3 40 58 PM" src="https://github.com/jamessimone/apex-rollup/assets/146372655/32c04a7d-03f3-4706-aa99-aca37d680ddb">

**Comment by jamessimone** [comment]
@marapetro it _should_ be working with:

- Grandparent relationship field path: `OpportunityContactRoles.Contact.Partner_Stage_2_Date__c`
- One to many grandparent fields: `OpportunityContactRole.OpportunityId`

You're saying you've tried that combination? Let me know!

**Comment by marapetro** [comment]
@jamessimone Thanks for getting back to me. Yes, I am still getting the same outcome with this set up:

<img width="1089" alt="Screenshot 2024-02-14 at 11 24 52 AM" src="https://github.com/jamessimone/apex-rollup/assets/146372655/0cf56afc-37e4-4097-a926-5c4658c55b8b">

Attaching below debug logs from the run initiated from the 'Recalculate rollups' button on the record.

Additionally, when I try to run only this rollup from the 'Recalculate rollup' tab the Queueable Job shows the same error as from the email above ('Developer script exception from Rippling : 'RollupFullBatchRecalculator' for job id '707RT000003f7wM' : Didn't understand relationship 'OpportunityContactRoles' in field path. If you are attempting to use a custom relationship, be sure to append the '__r' after the custom relationship name. Please reference your WSDL or the describe call for the appropriate names.')

I also tried removing the where statement and the outcome was the same.

Thanks for looking into this!
[apex-07LRT000001TxAf2AK.log](https://github.com/jamessimone/apex-rollup/files/14288530/apex-07LRT000001TxAf2AK.log)
[apex-07LRT000001TyeZ2AS.log](https://github.com/jamessimone/apex-rollup/files/14288531/apex-07LRT000001TyeZ2AS.log)

**Reply by jamessimone** [reply]
OK, I will take a look and let you know. Apologies that you're having an issue with this!

**Reply by marapetro** [reply]
Thank you for looking and no apology needed. :)
I can always do this via a formula field on OpportunityContactRole if it becomes urgent.

**Reply by marapetro** [reply]
Just a quick follow up to add another data point - I am getting the same error when trying to set grandfather path with other junction objects other than OpportunityContactRole.

![Screenshot 2024-02-15 at 11 10 30 AM](https://github.com/jamessimone/apex-rollup/assets/146372655/b41bebbb-33b1-4f02-b0f7-8851414dad23)


Error message:
'RollupFullBatchRecalculator' for job id '707RT000003hi0s' : Didn't understand relationship 'Customer_Insights__r' in field path. If you are attempting to use a custom relationship, be sure to append the '__r' after the custom relationship name. Please reference your WSDL or the describe call for the appropriate names.

**Comment by jamessimone** [comment]
@marapetro I am planning to take a look at this as soon as I can. I am wrapping up one other issue that has been reported at the moment, apologies for the slow responses thus far.

**Reply by jamessimone** [reply]
@marapetro I went to try to reproduce this issue today but was able to get your exact same scenario running without an issue:
![image](https://github.com/jamessimone/apex-rollup/assets/16430727/17786242-b198-4b8c-8c2c-7b2096450d34)

I don't see anything obviously different between my screenshot and yours, but maybe between the two of us we can narrow down where this is differing.

**Reply by marapetro** [reply]
Thanks for looking into this @jamessimone. I cannot figure out what's different either. I was concerned that the issue is specific to my current org, so I've don the following:

1. Signed up for a new dev org
2. Installed apex rollup package to that org
3. Turned on Custom Settings.
4. Created new field on a Contact (Max_Opp_Amount__c)
5. Created new Rollup CMDT record (screenshot below).
6. Still getting the same error (debug logs below).

![Screenshot 2024-02-19 at 12 18 57 PM](https://github.com/jamessimone/apex-rollup/assets/146372655/539c0cc2-b5f5-49bc-82d8-4b8436d1fb22)
[NewDevOrgRollupDebugLog.log](https://github.com/jamessimone/apex-rollup/files/14336607/NewDevOrgRollupDebugLog.log)

I am happy to get you access to my dev org if that would be helpful.
